packages/react-noop-renderer/src/ReactNoopFlightServer.js

```diff
@@ -54,17 +55,29 @@ const ReactNoopFlightServer = ReactFlightServer({
   },
 });
 
+type ServerContextJSONValue =
+  | string
+  | boolean
+  | number
+  | null
+  | $ReadOnlyArray<ServerContextJSONValue>;
+
 type Options = {
   onError?: (error: mixed) => void,
 };
 
-function render(model: ReactModel, options?: Options): Destination {
+function render(
+  model: ReactModel,
+  options?: Options,
+  context?: Array<[string, ServerContextJSONValue]>,
```

Comment:
This is not the format we're going with anymore, right?

It should probably go on the options object. Two optional arguments at the end gets awkward.

I think we'll actually need a different API for this because I've hit use cases where you want to have to different contexts for different subtrees but we can do that in a follow up.

Reply:
> This is not the format we're going with anymore, right?

Hmm when we spoke about it we said we were going with this format still, but that now end users would need to use reverse linked list of context nodes to create this format themselves and filter out duplicates however they like?

Reply:
Oh I was under the impression that the filtered format would still be this. So you have the option to just pass it straight through or convert it but if you convert it you’re responsible for converting it back again.

Reply:
Hmm, whats the right trade off? I think the array format is more compact right? It doesn't really matter which context is the `parent` of who so that would just be like wasted bytes?

Reply:
Also since the object is frozen in __DEV__ they'd need to create their own anyways?

Reply:
We can leave this decision until a follow up. We still have to figure out how to render multiple roots in one request too.